src/main/java/pingis/services/DataImporter.java

```diff
@@ -0,0 +1,223 @@
+import pingis.repositories.ChallengeRepository;
+import pingis.repositories.TaskImplementationRepository;
+import pingis.repositories.TaskRepository;
+import pingis.repositories.UserRepository;
+
+@Component
+public class DataImporter implements ApplicationRunner{
+
+    private String jsonString;
+    private ChallengeRepository cr;
+    private ChallengeImplementationRepository cir;
+    private TaskRepository tr;
+    private UserRepository ur;
+    private TaskImplementationRepository tir;
+    private HashMap<String, User> users = new HashMap();
+    private ArrayList<Challenge> challenges = new ArrayList();
+    private ArrayList<Task> tasks = new ArrayList();
+    private ArrayList<TaskImplementation> taskImplementations = new ArrayList();
+    private ArrayList<ChallengeImplementation> challengeImplementations = new ArrayList();
+    private int taskid = 0;
+//CHECKSTYLE:OFF
+
+    @Autowired
+    public DataImporter(ChallengeRepository cr, TaskRepository tr, UserRepository ur,
+            ChallengeImplementationRepository cir, TaskImplementationRepository tir) {
+        this.cr = cr;
+        this.tr = tr;
+        this.ur = ur;
+        this.cir = cir;
+        this.tir = tir;
+    }
+    
+    public DataImporter() {
+    }
+    
+    public void run(ApplicationArguments args) throws Exception {
+        DataLoaderIO io = new DataLoaderIO();
+        readData(io);
+        generateUsers();
+        generateEntities();
+        printResults();
+        populateDatabase();
+    }
+
+    public void readData(IO io) {
+        String out = "";
+        while(io.hasNext()) {
+            out += io.nextLine();
+        }
+        this.jsonString = out;
+    }
+
+    public String getJsonString() {
+        return jsonString;
+    }
+    
+    public void setJsonString(String jsonString) {
+        this.jsonString = jsonString;
+    }
+    
+    public void generateUsers() {
+        long userId = new Random().nextInt(Integer.MAX_VALUE);
+
+        
+        users.put("modeluser", new User(DataLoader.UserType.TMC_MODEL_USER.getId(), DataLoader.UserType.TMC_MODEL_USER.name(), 100)); 
+        users.put("testuser", new User(DataLoader.UserType.TEST_USER.getId(), DataLoader.UserType.TEST_USER.name(), 5));
+        users.put("impluser", new User(DataLoader.UserType.IMPLEMENTATION_USER.getId(), DataLoader.UserType.IMPLEMENTATION_USER.name(), 1));
+
+    }
+
+    public HashMap<String, User> getUsers() {
+        return users;
+    }
+
+    public ArrayList<Challenge> getChallenges() {
+        return challenges;
+    }
+
+    public ArrayList<Task> getTasks() {
+        return tasks;
+    }
+
+    public ArrayList<TaskImplementation> getTaskImplementations() {
+        return taskImplementations;
+    }
+
+    public ArrayList<ChallengeImplementation> getChallengeImplementations() {
+        return challengeImplementations;
+    }
+    
+    public void generateEntities() {
+        JSONObject obj = new JSONObject(this.jsonString);
```

Comment:
Why not use a JSON data binding library like Jackson here instead of parsing it manually?